### PR TITLE
[TS] add production deploy checklist

### PR DESCRIPTION
**Pull request by sw-yx**
## What does this PR do?

add production deploy checklist as agreed with Samar and @lorensr

### CODE REVIEW

**Comment by lorensr** [comment]
Here's a section to use in lieu of `## Prebuild the Worker`

https://gist.github.com/lorensr/56fb1b6980e2bc7138a663897473daab

**Comment by sw-yx** [comment]
merged your gist in

File: docs/content/how-to-develop-a-worker-program-in-typescript.md [hunk 1]
```diff
@@ -25,11 +25,13 @@ Activities are directly required and run by Workers in the Node.js environment.
...
 Workers are very flexible - you can host any or all of your Workflows and Activities on a Worker, and you can host multiple Workers in a single machine.
 
-There are three things the Worker needs:
+There are three main things the Worker needs:
 
-- `activities`: You import and supply these directly to the Worker.
-- `workflowsPath`: A path to your workflows file to pass to Webpack, e.g. `require.resolve('./workflows')`. Workflows will be bundled with their dependencies, which you can finetune with `nodeModulesPaths`.
-- `taskQueue`: the Task Queue to listen to.
+- `taskQueue`: the Task Queue to poll. This is the only required argument.
+- `activities`: You import and supply these directly to the Worker (optional if no )
```
Comment (lorensr) [inline docs/content/how-to-develop-a-worker-program-in-typescript.md:31]: `if no )`

**Review by lorensr — APPROVED** [review]

File: docs/content/how-to-develop-a-worker-program-in-typescript.md (hunk 1 above)
Comment (sw-yx) [reply docs/content/how-to-develop-a-worker-program-in-typescript.md:31]:
```suggestion
- `activities`: You import and supply these directly to the Worker (optional if no)
```

File: docs/content/how-to-develop-a-worker-program-in-typescript.md [hunk 2]
```diff
@@ -25,11 +25,13 @@ Activities are directly required and run by Workers in the Node.js environment.
...
 Workers are very flexible - you can host any or all of your Workflows and Activities on a Worker, and you can host multiple Workers in a single machine.
 
-There are three things the Worker needs:
+There are three main things the Worker needs:
 
-- `activities`: You import and supply these directly to the Worker.
-- `workflowsPath`: A path to your workflows file to pass to Webpack, e.g. `require.resolve('./workflows')`. Workflows will be bundled with their dependencies, which you can finetune with `nodeModulesPaths`.
-- `taskQueue`: the Task Queue to listen to.
+- `taskQueue`: the Task Queue to poll. This is the only required argument.
+- `activities`: You import and supply these directly to the Worker (optional if no)
```
Comment (sw-yx) [inline docs/content/how-to-develop-a-worker-program-in-typescript.md:31]:
```suggestion
- `activities`: Optional. Imported and supplied directly to the Worker. Not file path name. 
```

**Pull request merged**